webpack.embedding-sdk.config.js

```diff
@@ -0,0 +1,215 @@
+          resourceQuery: { not: [/component|source/] },
+        },
+        {
+          test: /\.css$/,
+          use: [
+            {
+              loader: "style-loader",
+            },
+            //   { loader: MiniCssExtractPlugin.loader },
+            { loader: "css-loader", options: CSS_CONFIG },
+            { loader: "postcss-loader" },
+          ],
+        },
+        // TODO: this should be enabled only in dev mode
+        {
+          test: /\.js$/,
+          exclude: /node_modules/,
+          enforce: "pre",
+          use: ["source-map-loader"],
+        },
+        {
+          test: /\.svg/,
+          type: "asset/source",
+          resourceQuery: /source/, // *.svg?source
+        },
+        {
+          test: /\.svg$/i,
+          issuer: /\.[jt]sx?$/,
+          resourceQuery: /component/, // *.svg?component
+          use: [
+            {
+              loader: "@svgr/webpack",
+              options: {
+                ref: true,
+              },
+            },
+          ],
+        },
+      ],
+    },
+
+    resolve: {
+      extensions: [
+        ".webpack.js",
+        ".web.js",
+        ".js",
+        ".jsx",
+        ".ts",
+        ".tsx",
+        ".css",
+        ".svg",
+      ],
+      alias: {
+        assets: ASSETS_PATH,
+        fonts: FONTS_PATH,
+        metabase: SRC_PATH,
+        "metabase-lib": LIB_SRC_PATH,
+        "metabase-enterprise": ENTERPRISE_SRC_PATH,
+        "metabase-types": TYPES_SRC_PATH,
+        cljs: isDevMode ? CLJS_SRC_PATH_DEV : CLJS_SRC_PATH,
+
+        style: SRC_PATH + "/css/core/index",
+
+        ace: __dirname + "/node_modules/ace-builds/src-noconflict",
+
+        // NOTE @kdoh - 7/24/18
+        // icepick 2.x is es6 by defalt, to maintain backwards compatability
+        // with ie11 point to the minified version
+        icepick: __dirname + "/node_modules/icepick/icepick.min",
+
+        // conditionally load either the EE plugins file or an empty file in the CE code tree
+        "ee-plugins":
+          process.env.MB_EDITION === "ee"
+            ? ENTERPRISE_SRC_PATH + "/plugins"
+            : SRC_PATH + "/lib/noop",
+        "ee-overrides":
+          process.env.MB_EDITION === "ee"
+            ? ENTERPRISE_SRC_PATH + "/overrides"
+            : SRC_PATH + "/lib/noop",
+      },
+    },
+
+    externals: {
+      react: "react",
+      "react-dom": "react-dom",
+      "react/jsx-runtime": "react/jsx-runtime",
+    },
+
+    optimization: !isDevMode
+      ? {
+          minimize: !shouldDisableMinimization,
+          minimizer: [
+            new TerserPlugin({
+              minify: TerserPlugin.swcMinify,
+            }),
+          ],
+        }
+      : undefined,
+
+    plugins: [
+      new NodePolyfillPlugin(), // for crypto, among others
+      // https://github.com/remarkjs/remark/discussions/903
+      new webpack.ProvidePlugin({
+        process: "process/browser.js",
+      }),
+      new ForkTsCheckerWebpackPlugin({
```

Comment:
I don't see a reason to use it instead of normal tsc at CI, but if you have arguments :+1:

Reply:
isn't tsc enough for it?